community/unsafe/src/main/java/org/neo4j/unsafe/impl/internal/dragons/UnsafeUtil.java

```diff
@@ -307,6 +307,32 @@ public static Object getAndSetObject( Object obj, long offset, Object newValue )
         return unsafe.getAndSetObject( obj, offset, newValue );
     }
 
+    /**
+     * Atomically exchanges provided <code>newValue</code> with the current value of field or array element, with
+     * provided <code>offset</code>.
+     */
+    public static long getAndSetLong( Object object, long offset, long newValue )
+    {
+        return unsafe.getAndSetLong( object, offset, newValue );
+    }
+
+    /**
+     * Atomically set field or array element to a maximum between current value and provided <code>newValue</code>
+     */
+    public static void compareAndSetMaxLong( Object object, long fieldOffset, long newValue )
+    {
+        long currentValue;
+        do
+        {
+            currentValue = UnsafeUtil.getLong( object, fieldOffset );
```

Comment:
I think this needs to be a `getLongVolatile`.

Reply:
yes, thx that's a mistake it should be